src/Testcontainers.ActiveMq/ArtemisContainer.cs

```diff
@@ -0,0 +1,31 @@
+namespace Testcontainers.ActiveMq;
+
+/// <inheritdoc cref="DockerContainer" />
+[PublicAPI]
+public sealed class ArtemisContainer : DockerContainer
+{
+    private readonly ActiveMqConfiguration _configuration;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ArtemisContainer" /> class.
+    /// </summary>
+    /// <param name="configuration">The container configuration.</param>
+    /// <param name="logger">The logger.</param>
+    public ArtemisContainer(ActiveMqConfiguration configuration, ILogger logger)
+        : base(configuration, logger)
+    {
+        _configuration = configuration;
+    }
+
+    /// <summary>
+    /// Gets the broker address.
+    /// </summary>
+    /// <returns>The broker address.</returns>
+    public string GetBrokerAddress()
+    {
+        var endpoint = new UriBuilder("tcp://", Hostname, GetMappedPublicPort(ArtemisBuilder.ArtemisMainPort));
+        endpoint.UserName = Uri.EscapeDataString(_configuration.Username);
+        endpoint.Password = Uri.EscapeDataString(_configuration.Password);
```

Comment:
Does it make sense to include the credentials in the connection string if the client library needs to be configured separately anyway?

Reply:
I was deliberating over this question.
There are libraries which are able to extract usernames and passwords from transport addresses, such as Microsoft.Azure.Amqp. I decided to include the credentials in the connection string, so that users who are using such libraries will be able to leverage them.

Reply:
Thanks, that makes sense.